demisto_sdk/commands/pre_commit/pre_commit_context.py

```diff
@@ -0,0 +1,202 @@
+import itertools
+import os
+import shutil
+from collections import defaultdict
+from dataclasses import dataclass, field
+from functools import cached_property
+from pathlib import Path
+from typing import Dict, List, Optional, Set, Tuple
+
+from demisto_sdk.commands.common.content_constant_paths import CONTENT_PATH
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.commands.common.tools import (
+    get_file_or_remote,
+    get_remote_file,
+    string_to_bool,
+)
+from demisto_sdk.commands.content_graph.objects.integration_script import (
+    IntegrationScript,
+)
+from demisto_sdk.commands.pre_commit.hooks.hook import Hook
+
+IS_GITHUB_ACTIONS = string_to_bool(os.getenv("GITHUB_ACTIONS"), False)
+
+PRECOMMIT_TEMPLATE_NAME = ".pre-commit-config_template.yaml"
+PRECOMMIT_TEMPLATE_PATH = CONTENT_PATH / PRECOMMIT_TEMPLATE_NAME
+PRECOMMIT_FOLDER = CONTENT_PATH / ".pre-commit"
+PRECOMMIT_CONFIG = PRECOMMIT_FOLDER / "config"
+PRECOMMIT_CONFIG_MAIN_PATH = PRECOMMIT_CONFIG / ".pre-commit-config-main.yaml"
+PRECOMMIT_DOCKER_CONFIGS = PRECOMMIT_CONFIG / "docker"
+
+
+@dataclass
+class PreCommitContext:
+    """This class is saving the context run of pre-commit hooks.
+    This data is shared between all hooks"""
+
+    input_files: Optional[List[Path]]
+    all_files: bool
+    mode: str
+    language_version_to_files_with_objects: Dict[
+        str, Set[Tuple[Path, Optional[IntegrationScript]]]
+    ]
+    run_hook: Optional[str] = None
+    skipped_hooks: Set[str] = field(default_factory=set)
+    run_docker_hooks: bool = True
+    dry_run: bool = False
+
+    def __post_init__(self):
+        """
+        We initialize the hooks and all_files for later use.
+        """
+        shutil.rmtree(PRECOMMIT_FOLDER, ignore_errors=True)
+        PRECOMMIT_FOLDER.mkdir()
+        PRECOMMIT_CONFIG.mkdir()
+        PRECOMMIT_DOCKER_CONFIGS.mkdir()
+
+        self.precommit_template = get_file_or_remote(PRECOMMIT_TEMPLATE_PATH)
+        remote_config_file = get_remote_file(str(PRECOMMIT_TEMPLATE_PATH))
+        if remote_config_file and remote_config_file != self.precommit_template:
+            logger.info(
+                f"Your local {PRECOMMIT_TEMPLATE_NAME} is not up to date to the remote one."
+            )
+        if not isinstance(self.precommit_template, dict):
+            raise TypeError(
+                f"Pre-commit template in {PRECOMMIT_TEMPLATE_PATH} is not a dictionary."
+            )
+        self.hooks = self._get_hooks(self.precommit_template)
+        self.hooks_need_docker = self._hooks_need_docker()
```

Comment:
print contents as json of this object to debug log

Reply:
added a print to debug